sw-services/src/main/java/org/egov/swservice/repository/builder/SWQueryBuilder.java

```diff
@@ -0,0 +1,197 @@
+package org.egov.swservice.repository.builder;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import org.apache.commons.lang.StringUtils;
+import org.egov.common.contract.request.RequestInfo;
+import org.egov.swservice.config.SWConfiguration;
+import org.egov.swservice.model.Property;
+import org.egov.swservice.model.SearchCriteria;
+import org.egov.swservice.util.SewerageServicesUtil;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import org.springframework.util.CollectionUtils;
+
+
+@Component
+public class SWQueryBuilder {
+
+	@Autowired
+	SewerageServicesUtil sewerageServicesUtil;
+
+	@Autowired
+	SWConfiguration config;
+
+	private static final String INNER_JOIN_STRING = "INNER JOIN";
+	 private static final String LEFT_OUTER_JOIN_STRING = " LEFT OUTER JOIN ";
+	//private static final String Offset_Limit_String = "OFFSET ? LIMIT ?";
+	
+	private final static String noOfConnectionSearchQuery = "SELECT count(*) FROM eg_sw_connection WHERE";
+	
+	private final static String SEWERAGE_SEARCH_QUERY = "SELECT conn.*, sc.*, document.*, plumber.*, sc.connectionExecutionDate,"
+			+ "sc.noOfWaterClosets, sc.noOfToilets,sc.proposedWaterClosets, sc.proposedToilets, sc.connectionType, sc.connection_id as connection_Id, conn.id as conn_id, conn.applicationNo, conn.applicationStatus, conn.status, conn.connectionNo, conn.oldConnectionNo, conn.property_id, conn.roadcuttingarea, conn.action, conn.adhocpenalty, conn.adhocrebate, conn.adhocpenaltyreason, conn.adhocpenaltycomment, conn.adhocrebatereason, conn.adhocrebatecomment,"
+			+ " conn.roadtype, document.id as doc_Id, document.documenttype, document.filestoreid, document.active as doc_active, plumber.id as plumber_id, plumber.name as plumber_name, plumber.licenseno,"
+			+ " plumber.mobilenumber as plumber_mobileNumber, plumber.gender as plumber_gender, plumber.fatherorhusbandname, plumber.correspondenceaddress, plumber.relationship FROM eg_sw_connection conn "
+	+  INNER_JOIN_STRING 
+	+" eg_sw_service sc ON sc.connection_id = conn.id"
+	+  LEFT_OUTER_JOIN_STRING
+	+ "eg_sw_applicationdocument document ON document.swid = conn.id" 
+	+  LEFT_OUTER_JOIN_STRING
+	+ "eg_sw_plumberinfo plumber ON plumber.swid = conn.id";
+
+	/**
+	 * 
+	 * @param criteria on search criteria
+	 * @param preparedStatement preparedStatement
+	 * @param requestInfo
+	 * @return
+	 */
+	
+	
+	private final String paginationWrapper = "SELECT * FROM " +
```

Comment:
Latest record should be written first. Refer to changes in TL in TL-RENEWAL-FEATURE branch